cmake/Utils.cmake

```diff
@@ -176,9 +175,17 @@ function(xgboost_set_cuda_flags target)
 
   set_target_properties(${target} PROPERTIES
     CUDA_STANDARD 17
-    CUDA_STANDARD_REQUIRED ON
-    CUDA_SEPARABLE_COMPILATION OFF
-    CUDA_RUNTIME_LIBRARY Static)
+    CUDA_STANDARD_REQUIRED ON)
+
+  if(CMAKE_INTERPROCEDURAL_OPTIMIZATION AND DEFINED CMAKE_CUDA_DEVICE_LINK_OPTIONS_IPO)
```

Comment:
This seems to be an internal flag of CMake, I can't find any document other than its internal use https://github.com/Kitware/CMake/blob/734a009dcf0b58e9b109e9aec20f61cbc6708504/Modules/Compiler/NVIDIA-CUDA.cmake#L17 .

Reply:
The `CMAKE_CUDA_DEVICE_LINK_OPTIONS_IPO` variable containing the LTO compiler flags for CUDA which isn't defined otherwis.  It seemed more clear than just checking for v3.25 without context but since it is documented in the release notes for v3.25 it's probably better to just check the CMAke version with a comment explaining that CUDA LTO was enabled in that CMake version.  I'll update accordingly.